### DISCUSSION TITLE
Response headers and caching

### DISCUSSION THREAD

**Discussion by Petervdburg** [miguelgrinberg/microdot#120] [Q&A] [answered]
Hi, my very basic server is serving some html (rendered template using microdot_utemplate) and a background image.

The image isnt caching in my browser (Chrome), and my reading suggests I put some headers into my response such as Cache-Control: max-age=31536000 but I have no idea how. I couldn't find any examples in the docs or examples.

Some code snippets:
`app = Microdot()`
`Response.default_content_type = 'text/html'
`
`@app.get('/water')`
`async def spa_temps(request):`
`
	return render_template('water7.html',temp = request.g.pool_temp)
`
Ideas?
Thanks

**Comment by miguelgrinberg** [comment]
The docs that apply to this are here: https://microdot.readthedocs.io/en/latest/intro.html#the-three-parts-of-a-response

How are you serving your image? Because the example that you posted is for your HTML, but you want to cache the image.

**Comment by miguelgrinberg** [comment, marked as answer]
I just realized that if you are using `send_file()` to serve your image, it wasn't easy to include a `Cache-Control` header. I have now added a `max_age` argument to `send_file()` that you can use to easily indicate what the cache duration is.

```python
@app.get('/image.jpg')
def image(request):
    return send_file('/static/image.jpg', max_age=31536000)
```

Hope this helps!

**Reply by Petervdburg** [reply]
Thanks for the quick response, Miguel and bang on. Apologies for not explaining things properly. It is the 'static' images associated with the html. Your patch now returns the 'Cache-Control: max-age=xxxx' in the Response and I see the browser retrieving the image from memory (now taking uSec vs seconds).
